temporal-sdk/src/main/java/io/temporal/internal/statemachines/UpdateProtocolStateMachine.java

```diff
@@ -53,7 +57,14 @@ enum State {
     NEW,
     REQUEST_INITIATED,
     ACCEPTED,
+    ACCEPTED_COMMAND_CREATED,
```

Comment:
I'll note I didn't remove the look ahead for protocol messages. It **may** be possible to with message commands, but it would cause the state machine to have two totally different paths depending on if it is replaying or not and it would need different special logic inside the `WorkflowStateMachine` to handle determinism and the event loop.

Reply:
yeah totally agree it isn't unusual, I just don't think it actually is better since patch handing SM keeps track of replay in the state machine and still needs look ahead.